src/javaServerStarter.ts

```diff
@@ -1,24 +1,48 @@
-import { workspace, ExtensionContext } from 'vscode'
+import { workspace, ExtensionContext, window } from 'vscode'
 import { Executable, ExecutableOptions } from 'vscode-languageclient';
 import { RequirementsData } from './requirements';
 import * as os from 'os';
 import * as path from 'path';
 import { xmlServerVmargs, getJavaagentFlag, getKey, IS_WORKSPACE_VMARGS_XML_ALLOWED, getXMLConfiguration } from './settings';
+import * as http from 'http';
+import * as fs from 'fs';
+import { createHash, Hash } from 'crypto';
 const glob = require('glob');
 
 declare var v8debug;
 
 const DEBUG = (typeof v8debug === 'object') || startedInDebugMode();
 
-export function prepareExecutable(requirements: RequirementsData, xmlJavaExtensions: string[], context: ExtensionContext): Executable {
+export async function prepareExecutable(requirements: RequirementsData, xmlJavaExtensions: string[], context: ExtensionContext): Promise<Executable> {
   let executable: Executable = Object.create(null);
   let options: ExecutableOptions = Object.create(null);
   options.env = process.env;
   options.stdio = 'pipe';
   executable.options = options;
-  executable.command = path.resolve(requirements.java_home + '/bin/java');
-  executable.args = prepareParams(requirements, xmlJavaExtensions, context);
-  return executable;
+  const useBinary = getXMLConfiguration().get("server.binary.enabled", false);
+  const javaExecutable: () => Executable = (): Executable => {
+    executable.command = path.resolve(requirements.java_home + '/bin/java');
+    executable.args = prepareParams(requirements, xmlJavaExtensions, context);
+    return executable;
+  }
+  if (useBinary) {
+    //TODO handle case when other extensions contributing jars expect to force JVM mode
+    // TODO: handle arguments to the binary
+    return getServerBinary().then((serverBinary: string): Executable => {
+      const hash: Hash = createHash('sha256');
+      hash.update(fs.readFileSync(path.resolve(serverBinary)));
+      const hashDigest = hash.digest('hex').toLowerCase();
+      const expectedHashPath = path.resolve(__dirname, '../server', `lemminx-${os.platform}-hash`);
+      const expectedHash = fs.readFileSync(expectedHashPath).toString('utf-8').split(' ')[0];
```

Comment:
Need to handle errors like 
stack trace: Error: ENOENT: no such file or directory, open '/Users/fbricon/Dev/projects/vscode-xml/server/lemminx-darwin-hash'
	at Object.openSync (fs.js:440:3)
	at Object.func [as openSync] (electron/js2c/asar.js:140:31)
	at Object.readFileSync (fs.js:342:35)
	at Object.readFileSync (electron/js2c/asar.js:542:40)
	at /Users/fbricon/Dev/projects/vscode-xml/dist/extension.js:16298:41

Reply:
If the hash is missing, the user gets a warning asking them if they want to continue.